### PR TITLE
feat: add side panel

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
New component that displays a menu within a panel

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: .storybook/head.html [hunk 1]
```diff
@@ -1,6 +1,4 @@
 <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+<link rel="stylesheet" href="https://talend.github.io/bootstrap-theme/bootstrap.css">
```
Comment (frassinier) [inline .storybook/head.html]: It could be cool to add it from another PR but why not

File: src/AppHeaderBar/AppHeaderBar.scss [hunk 2]
```diff
@@ -24,5 +28,5 @@ $tlnd-app-menu-txt-color: white !default;
 	padding: 12px 20px 0;
 }
 .logo {
-  max-width: '100px';
+	max-width: 100px;
```
Comment (frassinier) [inline src/AppHeaderBar/AppHeaderBar.scss]: Thanks, but another PR for that too :)

File: src/SidePanel/SidePanel.component.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import classnames from 'classnames';
+import { Button } from 'react-bootstrap';
+import { themr } from 'react-css-themr';
+import { SIDE_PANEL } from '../identifiers';
+
+/**
+ * This component aims to display links as a menu.
+ * @param {object} props react props
```
Comment (frassinier) [inline src/SidePanel/SidePanel.component.js]: Add example impl here please

File: src/SidePanel/SidePanel.component.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	return (
+		<nav className={navClassName}>
+			<Button className={ theme.toggleBtn }
+			        bsStyle="link"
+			        onClick={ props.onToggleDock }
+			        aria-hidden="true"
+			        title="Toggle side panel">
+				<i className="fa fa-arrow-left"/>
+			</Button>
```
Comment (frassinier) [inline src/SidePanel/SidePanel.component.js]: Cool stuff :+1:

File: src/SidePanel/SidePanel.scss [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	padding: 0 $side-panel-spacing;
+	overflow-x: hidden;
+	transition: .2s padding ease-out;
+}
+
+.actionsList {
+	margin: 0;
+	padding: 0;
+	list-style: none;
+	margin: 0 -1*($side-panel-spacing);
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: sass-lint will throw some warn here

File: stories/SidePanel.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+			{ label: 'Preparations', icon: 'fa fa-asterisk', onClick: action('Preparations clicked') },
+			{ label: 'Datasets', icon: 'fa fa-file-excel-o', onClick: action('Datasets clicked') },
+			{ label: 'Favorites', icon: 'fa fa-star', onClick: action('Favorites clicked') }
+		];
+
+		return (
+			<div>
+				<h1>AppHeaderBar</h1>
+				<h2>Definition</h2>
+				<a href="http://guidelines.talend.com/document/92132#/ui-controls/app-header">Spec</a>
```
Comment (frassinier) [inline stories/SidePanel.js]: Wrong link

File: stories/SidePanel.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
```
Comment (frassinier) [inline stories/SidePanel.js]: Wrong link

File: stories/SidePanel.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { storiesOf, action } from '@kadira/storybook';
+
+import { SidePanel } from '../src/index';
+
+storiesOf('Side Panel', module)
+	.add('displayed', () => {
```
Comment (frassinier) [inline stories/SidePanel.js]: `.add('default', ...` ?

File: stories/AppHeaderBar.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { storiesOf, action, linkTo } from '@kadira/storybook';
```
Comment (frassinier) [inline stories/AppHeaderBar.js]: Must be another PR for that too :)

File: stories/SidePanel.js (hunk 6 above)
Comment (jsomsanith) [inline stories/SidePanel.js]: I was washing my dishes when I suddenly had this conversation with my brain : 
Brain : "Hey Jimmy, you completely messed up with the story, copy/paste and you forgot to change the rest."
Me : "Oh, you're right, you're the best brain I ever had ! I'll just rework that before anyone review it"

And I ended up here, seing that you @frassinier had already reviewed it. What a hard job with those people that go faster than you.

Comment (jsomsanith) [inline stories/SidePanel.js]: It's too late, so I'll jut rework all your review tomorrow morning :)

File: src/SidePanel/SidePanel.component.js (hunk 4 above)
Comment (jsomsanith) [inline src/SidePanel/SidePanel.component.js]: Thanks but that's not me, I just took it from previous implementation

**Comment by jsomsanith** [comment]
Review changes pushed. Please review/approve :)

File: src/SidePanel/SidePanel.scss [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+		margin: $side-panel-spacing 0;
+		line-height: 0.75;
+		color: $side-panel-toggle-color;
+		box-shadow: -2px 0 0 $side-panel-toggle-color;
+
+		&:hover,
+		&:focus,
+		&:active {
+			color: inherit;
+			text-decoration: none;
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: Why do you need it?

File: src/SidePanel/SidePanel.scss [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+			right: 0;
+			bottom: 0;
+			left: 0;
+			margin: -1rem;
+			border-radius: 50%;
+		}
+	}
+
+	.link {
+		color: inherit;
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: Why do you need it?

File: src/SidePanel/SidePanel.scss [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+		i {
+			padding-right: 20px;
+		}
+
+		span {
+			transition: 0.2s opacity linear;
+		}
+	}
+
+	.docked {
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: Should be `&.docked`

File: src/SidePanel/SidePanel.scss [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+/// @type Number (Unit)
+$side-panel-docked-spacing: 15px !default;
+
+.side-panel {
+	overflow-x: hidden;
+	padding: 0 $side-panel-spacing;
+	transition: 0.2s padding ease-out;
+
+	.action-list {
+		list-style: none;
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: Why do you need it?

File: src/SidePanel/SidePanel.scss [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+		margin: 0 (-$side-panel-spacing);
+		padding: 0;
```
Comment (frassinier) [inline src/SidePanel/SidePanel.scss]: Why do you need it?
